rts/Rendering/Textures/TextureAtlas.h

```diff
@@ -40,9 +38,27 @@ class CTextureAtlas
 	};
 
 public:
-	CTextureAtlas(unsigned int allocType = ATLAS_ALLOC_LEGACY);
+	CTextureAtlas(unsigned int allocType = ATLAS_ALLOC_LEGACY, const int atlasSizeX_ = 0, const int atlasSizeY_ = 0, std::string name_ = "");
+
+	CTextureAtlas(CTextureAtlas&& ta) noexcept
+	{
+		MoveFields(std::forward<CTextureAtlas>(ta));
```

Comment:
The usual trick for move-constructors is to write `*this = std::move(ta);`. That way you don't need a separate `MoveFields`, just the move-assignment operator.